staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -136,8 +136,8 @@ func getNestedInt64(obj map[string]interface{}, fields ...string) int64 {
 }
 
 func getNestedInt64Pointer(obj map[string]interface{}, fields ...string) *int64 {
-	if str, ok := getNestedField(obj, fields...).(*int64); ok {
```

Comment:
this works as long as our number-preserving json unmarshaler is used (stdlib turns numbers into float64)

also, getNestedSlice and getNestedMap assume values are strings and silently drop non-strings... the names should at least be updated so callers know their assumptions